src/generic/grid.cpp

```diff
@@ -11337,4 +11400,640 @@ wxGetContentRect(wxSize contentSize,
+    else
+        *role = wxROLE_SYSTEM_CELL;
+
+    return wxACC_OK;
+}
+
+// Returns a state constant.
+wxAccStatus wxGridCellAccessible::GetState(int childId, long* state)
+{
+    wxGrid* grid = wxDynamicCast(GetWindow(), wxGrid);
+    wxCHECK( grid, wxACC_FAIL );
+    if ( childId != wxACC_SELF )
+        return wxACC_FAIL;
+
+    long st = 0;
+
+    if ( !grid->IsEnabled() )
+        st |= wxACC_STATE_SYSTEM_UNAVAILABLE;
+
+    if ( !m_isColHeader && !grid->IsRowShown(m_row) )
+        st |= wxACC_STATE_SYSTEM_INVISIBLE;
+    else if ( !m_isRowHeader && !grid->IsColShown(m_col) )
+        st |= wxACC_STATE_SYSTEM_INVISIBLE;
+
+    if( grid->IsFocusable() )
+        st |= wxACC_STATE_SYSTEM_FOCUSABLE;
+
+    st |= wxACC_STATE_SYSTEM_SELECTABLE;
+
+    const int cursorRow = grid->GetGridCursorRow();
+    const int cursorCol = grid->GetGridCursorCol();
+
+    if ( !m_isColHeader && !m_isRowHeader && grid->HasFocus() && m_col==cursorCol && m_row==cursorRow)
+        st |= wxACC_STATE_SYSTEM_FOCUSED;
+
+    // check selection and visibility
+    if ( m_isRowHeader && m_isColHeader )
+    {
+        // check whether complete grid is selected?
+    }
+    else if ( m_isRowHeader )
+    {
+        wxArrayInt selectedRows = grid->GetSelectedRows();
+        if ( selectedRows.Index(m_row) != wxNOT_FOUND )
+            st |= wxACC_STATE_SYSTEM_SELECTED;
+    }
+    else if ( m_isColHeader )
+    {
+        wxArrayInt selectedCols = grid->GetSelectedCols();
+        if ( selectedCols.Index(m_col) != wxNOT_FOUND )
+            st |= wxACC_STATE_SYSTEM_SELECTED;
+    }
+    else
+    {
+        if ( grid->IsInSelection(m_row, m_col) )
+            st |= wxACC_STATE_SYSTEM_SELECTED;
+        if ( grid->IsVisible(m_row, m_col, false) )
+            st |= wxACC_STATE_SYSTEM_OFFSCREEN;
+        if ( grid->IsReadOnly(m_row, m_col) )
+            st |= wxACC_STATE_SYSTEM_READONLY;
+    }
+
+    *state = st;
+    return wxACC_OK;
+}
+
+// Returns a localized string representing the value for the object or child.
+wxAccStatus wxGridCellAccessible::GetValue(int childId, wxString* strValue)
+{
+    wxGrid* grid = wxDynamicCast(GetWindow(), wxGrid);
+    wxCHECK( grid, wxACC_FAIL );
+
+    if ( childId != wxACC_SELF )
+        return wxACC_FAIL;
+    // GetName now returns coordinate and value info at the same time
+    return wxACC_NOT_IMPLEMENTED;
+}
+
+// Gets the window with the keyboard focus.
+// If childId is 0 and child is nullptr, no object in this subhierarchy has the focus.
+// If this object has the focus, child should be 'this'.
+wxAccStatus wxGridCellAccessible::GetFocus(int* childId, wxAccessible** child)
+{
+    wxGrid* grid = wxDynamicCast(GetWindow(), wxGrid);
+    wxCHECK( grid, wxACC_FAIL );
+
+    *childId = wxACC_SELF;
+
+    const int cursorRow = grid->GetGridCursorRow();
+    const int cursorCol = grid->GetGridCursorCol();
+
+    if ( !m_isColHeader && !m_isRowHeader && cursorRow == m_row && cursorCol == m_col )
```

Comment:
Comparisons like this would be simpler to write if we used `wxGridCellCoords`, e.g. replacing `m_row` and `m_col` with a single `m_coords` would allow to reduce this to just

```suggestion
    if ( GetGridCursorCoords() == m_coords )
```

Reply:
Thanks. Done. This required several changes.
I hope I did not introduce bugs.

I hope this is correct to retain the previous coordinates for initialization of a new wxGridCellAccessible instance:
```
            wxGridCellCoords previous_coords = m_gridCellAccessible->m_coords;
            delete m_gridCellAccessible;
            m_gridCellAccessible = new wxGridCellAccessible(grid, childId);
            if ( m_gridCellAccessible->m_coords.GetRow() == previous_coords.GetRow() )
```